### PR TITLE
Proactive connection management refactoring

### PR DESCRIPTION
**Pull request by jeet1995**
This PR has the following objectives:
- Introduces a class `ContainerDirectConnectionMetadata` which will encapsulate all container-specific connection-related configuration.
- Removes an extra `checkArgument` call to verify the correctness of the `CosmosContainerProactiveInitConfig` built.
- Wires the JVM configuration - `COSMOS.MIN_CONNECTION_POOL_SIZE_PER_ENDPOINT` to `DirectConnectionConfig`. The objective here is to default to the JVM configuration and also make way for a public API to configure it through `DirectConnectionConfig` possibly in the future.
- Breaking change: Modifies the JVM configuration - `COSMOS.DEFENSIVE_WARMUP_CONCURRENCY` to `COSMOS.OPEN_CONNECTIONS_CONCURRENCY`

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java [hunk 1]
```diff
@@ -1048,7 +1048,6 @@ private void validateConfig() {
             Preconditions.checkArgument(preferredRegions != null, "preferredRegions cannot be null when proactiveContainerInitConfig has been set");
             Preconditions.checkArgument(this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount() <= this.preferredRegions.size(), "no. of regions to proactively connect to " +
                     "cannot be greater than the no.of preferred regions");
-            Preconditions.checkArgument(this.proactiveContainerInitConfig.getProactiveConnectionRegionsCount() > 0, "no. of proactive connection regions should be greater than 0");
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1051]: SHould we avoid negative values? I guess negative means on-demand connections only? Fine by me

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks!

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1051]: I think we should avoid negative values. Negative region configuration resulting in implicit disabling of proactive connection management is not very transparent.

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1051]: Agreed - so, why did you remove the check?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1051]: Will merge for now - we can readd the check in subsequent PR

**Comment by FabianMeiswinkel** [comment]
CI failures due to known issue with CI pipeline

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:1051]: When calling `CosmosProactiveContainerInitConfigBuilder#setProactiveConnectionRegionsCount` we have a negativity check. If this method is not called, the default region count is set to 1. A check in `CosmosClientBuilder` is redundant.

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**